src/main/java/com/aws/greengrass/builtin/services/configstore/ConfigStoreIPCEventStreamAgent.java

```diff
@@ -327,14 +338,17 @@ private void validateRequest(UpdateConfigurationRequest request) {
                         + restrictedConfigurationFields);
             }
 
-            Topics serviceTopics = kernel.findServiceTopic(serviceName);
-            if (serviceTopics == null) {
-                throw new InvalidArgumentsError("Component config not found for component " + serviceName);
+            final String targetServiceName =
+                    request.getComponentName() == null ? serviceName : request.getComponentName();
+
+            Topics targetServiceTopics = kernel.findServiceTopic(targetServiceName);
+            if (targetServiceTopics == null) {
+                throw new InvalidArgumentsError(String.format(COMPONENT_NOT_FOUND_ERROR_FORMAT, targetServiceName));
```

Comment:
switch this to `ResourceNotFound` if the updateConfiguration API is allowed to throw that type.

Reply:
It is not- allowed types are:
                 ServiceError, 
                 UnauthorizedError, 
                 ConflictError, 
                 FailedUpdateConditionCheckError, 
                 InvalidArgumentsError

Reply:
Thank you for checking